torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,161 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    is_alias_to_base: bool = False
+
+
+def serialize_views_meta(
+    arg_names, arg_types, input_kwargs, output_kwargs, arg_to_base_index
+):
+    def serialize_single_view(prefix, tensor, base_index):
```

Comment:
btw, when we serialize these, we need to check that e.g. `_base0_size` was not in the original kwargs. Otherwise we'd be clobbering it.

Reply:
done